app/scripts/directives/processTemplate.js

```diff
@@ -225,7 +229,7 @@
         };
         ctrl.template.labels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries(ctrl.labels));
 
-        DataService.create("processedtemplates", null, ctrl.template, context).then(
+        DataService.create(processedTemplatesVersion, null, ctrl.template, context).then(
```

Comment:
Not sure if this is correct, actually.  Should probably extract version from `ctrl.template` to ensure they match.

@spadgett ?

Reply:
Hmm. maybe is fine.  difference would be:
`{ group: "template.openshift.io", resource:"processedtemplates", version:"v1" }`
vs
`{ group: "template.openshift.io", resource:"templates", version:"v1" }`
if I switched to `APIService.objectToResourceGroupVersion(ctrl.template)`.  That doesn't seem correct.

Reply:
Using `preferredVersion` (first in above comment) causes:

![screen shot 2018-01-04 at 12 23 45 pm](https://user-images.githubusercontent.com/280512/34575926-4b966c00-f14a-11e7-8d70-3f6d49b165f2.png)

Using `templateRGV` (second in above) causes:

![screen shot 2018-01-04 at 12 23 09 pm](https://user-images.githubusercontent.com/280512/34575881-2cee2018-f14a-11e7-847d-2f403e6c8a31.png)

Kind of interesting, update to either fails.  Not sure why `namespace` is brought in, this shouldn't alter the `context` object, diagnosing.

Reply:
Hm, it looks need to use the API version from the template itself. I wonder if we have a bug in 3.7 if the template is declared using the API group.

Reply:
Resource should be `processedtemplates`, otherwise you're creating a template, not processing one.

Reply:
Think its a bug. If I used the `preferredVersion` but clear the `group`:

```
{
  group: "",  // drop the 'template.openshift.io'
  resource: "processedtemplates",
  version: "v1"
}
```
Works fine.

Reply:
Take that back, some do, some don't.  `cakephp` won't work.  `nodejs + mongo` works.  So depends on the template.

Reply:
Running `oc new-app -f origin/examples/quickstarts/cakephp-mysql.json  --loglevel=8` gives me a little insight.  POST to `https://127.0.0.1:8443/apis/template.openshift.io/v1/namespaces/baz/processedtemplates`, and apparently it updates the request body itself from `apiVersion: v1` to `"apiVersion":"template.openshift.io/v1"`.  

I'm not sure we want to be meddling with the templates directly in the console tho?